# Statistics Basic-2 Assignment Questions

### Q1. What are the three measures of central tendency?

Mean, Median and Mode

### Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

Median, mean, and mode are all measures of central tendency.

* `Mean:` Mean is the average of all values in a dataset. It is calculated by adding all the values and then dividing it by the lengths of the values. 

* `Median:` Median is the middle value in a dataset. If the dataset has an odd number of values, the median is the middle value. If the dataset has an even number of values, the median is the average of the two middle values.

* `Mode:` Mode is the most common value  in a dataset. A dataset can have one mode, more than one mode. The mode is useful for categorical data.

These measures calculate a dataset's central tendency. The choice of measure depends on the type of data. The mean is used when data is normally distributed and doesn't have extreme values. It's better to use the median if the data has outliers or is skewed. Mode lets you see the most common value in a dataset.

### Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [2]:
import numpy as np
import scipy.stats as stats

In [3]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [6]:
np.mean(data)

177.01875

In [7]:
np.median(data)

177.0

In [9]:
mode = stats.mode(data, keepdims=True)
mode.mode[0]

177.0

### Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [10]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [11]:
np.std(data)

1.7885814036548633

### Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

Measures of dispersion are used to describe the spread of a dataset. Range, variance, and standard deviation are three common measures of dispersion.

1. Range is the simplest measure of dispersion. It's the difference between the biggest and smallest values.

2. Variance is a more complex measure of dispersion. Basically, it measures how far the data points are from the mean.

3. Standard deviation is the square root of variance. It represents the spread of the data points from the mean.

In [18]:
data = [165, 170, 175, 178, 180, 182, 185, 188, 190, 193]
np.var(data)

71.24

In [17]:
np.ptp(data)

28

In [19]:
np.std(data)

8.440379138403676

The following is an example of how dispersion can be used to describe data spread. The following data set shows 10 people's heights in centimeters:

[165, 170, 175, 178, 180, 182, 185, 188, 190, 193]

The data set range is 193 - 165 = 28, variance is 71.24 and the deviation is 8.440379138403676.

### Q6. What is a Venn diagram?

Venn diagrams show logical relations between sets. Venn diagrams are powerful tools for visualizing data and understanding logical relationships. In a short, clear and concise way, they can explain complex information.



### Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
* A ∩ B
* A ⋃ B

A ∩ B = {2,6} and 
A U B = {0, 2, 3, 4, 5, 6, 7, 8, 10}

### Q8. What do you understand about skewness in data?

Skewness is a measure of distribution. A distribution is skewed if it is not symmetrical around its mean. There are two types of skewness: 

* Positive skewness occurs when the tail of the distribution is longer on the right side of the mean. 

* Negative skewness occurs when the tail of the distribution is longer on the left side of the mean.

Skewness can be measured using several methods. One common method is to use the Skewness coefficient. It shows us how much skewness is present in a distribution. A skewness coefficient of 0 indicates symmetrical distribution. A skewness coefficient greater than 0 shows positive skewness while less than 0 indicates negative skewness.

### Q9. If a data is right skewed then what will be the position of median with respect to mean?

In a right-skewed distribution, the mean is typically greater than the median.
Here is an example:

In [20]:
data = [1, 2, 2, 3, 3, 4, 5, 6, 8, 10]

In [21]:
np.mean(data)

4.4

In [22]:
np.median(data)

3.5

### Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

The main difference between covariance and correlation is that covariance measures the strength and direction of the relationship between two variables. In contrast, correlation is a measure of the relationship strength only.

In statistical analysis, covariance and correlation measure relationships between variables. Also, they predict the value of one variable based on another.

### Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

The formula is, $x̄ = \frac{\sum xi}{n}$

In [25]:
data = [1, 2, 3, 4, 5]

In [26]:
x̄ = (1 + 2 + 3 + 4 + 5) / 5

In [27]:
x̄

3.0

### Q12. For a normal distribution data what is the relationship between its measure of central tendency?

In a normal distribution, the mean, median, and mode are all equal. It's because the normal distribution has a symmetrical shape, so the data is evenly distributed.

Mean is the average of the data points, median is the middle value, and mode is the most frequent value.

In a normal distribution, the relationship between the mean, median, and mode helps us identify the center quickly. We can use this information to make inferences about the data, like the probability of a particular value.

### Q13. How is covariance different from correlation?

Basically, covariance measures whether two variables are moving in the same direction and how strong the relationship is, whereas correlation measures how strong the relationship is.

### Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

An outlier is a data point that's significantly different from the rest. They can affect central tendency and dispersion in many ways.

* Mean: The mean is the average of all the data points in a set. Having outliers can make the mean less representative of the overall set.

* Median: The median is the middle value of all the data points in a set. If an outlier isn't in the middle, it doesn't affect the median.

* Mode: The mode is the most frequent value in a set. A high frequency outlier can increase the mode.

* Range: The range is the difference between the largest and smallest values in a set. The range can be increased if there are outliers.

* Variance: The variance is a measure of how spread out the values in a set are. When outliers are in the set, they can make the other values more distant from each other. 

* Standard Deviation : We can calculate standard deviation by taking the square root of the variance of a set of values. Standard deviation and variance can both be affected by outliers.

#### Data with outlier

In [28]:
data = [1, 2, 3, 4, 5, 100]

In [29]:
np.mean(data)

19.166666666666668

In [30]:
np.median(data)

3.5

In [31]:
mode = stats.mode(data, keepdims=True)
mode.mode[0]

1

In [32]:
np.ptp(data)

99

In [33]:
np.var(data)

1308.4722222222222

In [34]:
np.std(data)

36.17281053805775

#### Data without outlier

In [35]:
data = [1, 2, 3, 4, 5, 6]

In [36]:
np.mean(data)

3.5

In [37]:
np.median(data)

3.5

In [38]:
mode = stats.mode(data, keepdims=True)
mode.mode[0]

1

In [39]:
np.ptp(data)

5

In [40]:
np.var(data)

2.9166666666666665

In [41]:
np.std(data)

1.707825127659933